wordninja.py

```diff
@@ -49,7 +59,20 @@ def best_match(i):
     while i>0:
         c,k = best_match(i)
         assert c == cost[i]
-        out.append(s[i-k:i])
+        # Apostrophe and digit handling (added by Genesys)
+        newToken = True
+        if not s[i-k:i] == "'": # ignore a lone apostrophe
+            if len(out) > 0:
+                # re-attach split 's and split digits
+                if out[-1] == "'s" \
+                  or (s[i-1] in '0123456789' and out[-1][0] in '0123456789'): # digits followed by digits
```

Comment:
you can use the isdigit() method of the string class here I think

Reply:
Thanks, I'll change that. A timeit experiment shows that my approach takes about 6 times as long as isdigit().